pkg/machines/reducers.es6

```diff
@@ -175,26 +175,72 @@ function vms(state, action) {
             }
             const updatedVm = Object.assign(indexedVm.vmCopy, {
                 lastMessage: action.payload.message,
-                lastMessageDetail: action.payload.detail});
+                lastMessageDetail: action.payload.detail,
+            });
 
             return replaceVm({ state, updatedVm, index: indexedVm.index });
         }
-        case 'UNDEFINE_VM':
-        {
+        case 'UNDEFINE_VM': {
             return state
                 .filter(vm => (action.connectionName !== vm.connectionName || action.name != vm.name ||
-                               (action.transientOnly && vm.persistent)));
+                    (action.transientOnly && vm.persistent)));
         }
-        case 'DELETE_UNLISTED_VMS':
-        {
+        case 'DELETE_UNLISTED_VMS': {
             return state
-                .filter(vm => (action.connectionName !== vm.connectionName || action.vmNames.indexOf(vm.name) >= 0) );
+                .filter(vm => (action.connectionName !== vm.connectionName || action.vmNames.indexOf(vm.name) >= 0));
         }
         default: // by default all reducers should return initial state on unknown actions
             return state;
     }
 }
 
+function osInfoList(state, action) {
+    state = state ? state : [];
+
+    logDebug('reducer osInfoList: action=' + JSON.stringify(action));
```

Comment:
every action passes this reducer. Please do debug logging just once or better remove before merge completely.